skills/src/main/java/cm/aptoide/skills/usecase/PayTicketUseCase.kt

```diff
@@ -8,42 +8,91 @@ import cm.aptoide.skills.BuildConfig
 import cm.aptoide.skills.repository.TicketRepository
 import cm.aptoide.skills.util.EskillsUri
 import io.reactivex.Single
+import java.math.BigDecimal
 
 class PayTicketUseCase(private val ticketRepository: TicketRepository) {
 
-  fun payTicket(ticketId: String, callbackUrl: String, productToken: String,
-                eskillsUri: EskillsUri, fragment: Fragment): Single<Any> {
+  fun payTicket(
+    ticketId: String, callbackUrl: String, productToken: String,
+    ticketPrice: BigDecimal, priceCurrency: String, eskillsUri: EskillsUri, fragment: Fragment
+  ): Single<Any> {
     return Single.fromCallable {
-      if (eskillsUri.getEnvironment() == EskillsUri.MatchEnvironment.LIVE) {
-        launchPurchaseFlow(eskillsUri, callbackUrl, ticketId, productToken, fragment)
+      val environment = eskillsUri.getEnvironment()
+      if (environment == EskillsUri.MatchEnvironment.LIVE || environment == null) {
+        launchPurchaseFlow(
+          eskillsUri,
+          callbackUrl,
+          ticketId,
+          productToken,
+          ticketPrice,
+          priceCurrency,
+          fragment
+        )
       }
       0
     }
   }
 
-  private fun launchPurchaseFlow(eskillsUri: EskillsUri, callbackUrl: String, ticketId: String, productToken: String, fragment: Fragment) {
+  private fun launchPurchaseFlow(
+    eskillsUri: EskillsUri,
+    callbackUrl: String,
+    ticketId: String,
+    productToken: String,
+    ticketPrice: BigDecimal,
+    priceCurrency: String,
+    fragment: Fragment
+  ) {
     val url: String =
-        (BACKEND_HOST + "/transaction/inapp?product=" + eskillsUri.getProduct() +
-            "&value=" + eskillsUri.getPrice() + "&currency=" + eskillsUri.getCurrency() +
-            "&domain=" + eskillsUri.getPackageName() + "&callback_url=" + callbackUrl +
-            "&order_reference=" + ticketId + "&product_token=" + productToken + "&skills")
+      (BACKEND_HOST + "/transaction/inapp?domain=" + eskillsUri.getPackageName() + "&callback_url=" + callbackUrl +
+          "&order_reference=" + ticketId + "&product_token=" + productToken + "&skills") + getOptionalFields(
+        eskillsUri, ticketPrice, priceCurrency
+      )
 
     val i = Intent(Intent.ACTION_VIEW)
     i.data = Uri.parse(url)
     i.setPackage(IAB_BIND_PACKAGE)
 
     val intent =
         PendingIntent.getActivity(fragment.requireContext()
-            .getApplicationContext(), 0, i,
-            PendingIntent.FLAG_UPDATE_CURRENT)
+          .getApplicationContext(), 0, i,
+          PendingIntent.FLAG_UPDATE_CURRENT
```

Comment:
Studio gives me this error `Missing PendingIntent mutability flag` you know what is it about?

Reply:
Not quite sure, apparently PendingIntent.FLAG_IMMUTABLE was introduced in api level 23. My studio doesn't show no error though. I've changed the code to prevent the situation anyway.